src/routes/(app)/c/private/[channel_id]/+page.server.ts

```diff
@@ -0,0 +1,32 @@
+import { getDb } from '$lib/server';
+import { getChannelById } from '$lib/server/services/channels';
+import { getPosts } from '$lib/server/services/content';
+import { error } from '@sveltejs/kit';
+import type { PageServerLoad } from './$types';
+import { canViewChannel } from '$lib/server/services/channels';
+import { assertAuth } from '$lib/server/auth';
+
+export const load: PageServerLoad = (async (event) => {
+    const db = await getDb();
+    const { channel_id } = event.params;
+    const data = await getChannelById(db, channel_id);
+    const channel_name = data.name;
+
+    assertAuth(event);
```

Comment:
perhaps move the auth assertion to the beginning of the function to avoid the potentially unnecessary computation above. We get to save a DB query in the case that a user is not authenticated if so.